docs/tutorials/migrating-your-ga4-data-to-snowplow/index.md

```diff
@@ -0,0 +1,227 @@
+...
+```
+- Some Snowplow fields have been pre-extracted from the GA4 data to make the SQL more readable, and to make using these fields in further calculations easier. These are:
+    - `domain_sessionidx`
+    - `page_url`
+    - `page_title`
+    - `page_referrer`
+    - `mkt_medium`
+    - `mkt_source`
+    - `mkt_term`
+    - `mkt_content`
+    - `mkt_campaign`
+    - `domain_sessionid`
+    
+As such, when writing SQL in the mapping, you can use these fields directly, e.g.:
+| Snowplow_field | mapping_sql |
+| --------------- | ----------- |
+| page_urlquery   | regexp_extract(page_url, r'(?:.*)\\?(.*)')|
+- If you wish to use a parameter from `event_params`, we have a UDF that will extract the parameter from the event_params column. The UDF is called `ga4_migrator_parse_params`. To use it, the SQL must be written in the following format:
+```sql
+JSON_VALUE(ga4_migrator_parse_params(event_params), '$.parameter_to_extract')
+```
+E.g.:
+```sql
+JSON_VALUE(ga4_migrator_parse_params(event_params), '$.page_title')
+```
+- When creating or updating any of the mapping fields and using a dbt package, it must be written in the following format, including the curly braces and the package reference: `{{dbt_package.function('column_name')}}`, e.g. `{{dbt_utils.get_url_host('page_referrer')}}`.
+
+- In the mapping table, there are two fields that should not be altered:\
+`geo_country_temp` and `geo_region_temp`. These are required for the `geo_country` and `geo_region` fields to be populated correctly.
+
+### Step 4 - Transform data
+
+Click the `Run transformation` button to run the migration. Behind the scenes, this sends a request to the API, which will run dbt. dbt will create the UDF that extracts the event_params, seed two tables that are used for geographical mapping, and create the final table that contains the migrated data.\
+
+### Step 5 - Import to Snowflake
+
+This step is optional, and is only if you wish to import the data into Snowflake. If you do not wish to import the data, you can skip this step.\
+Provide the following parameters:
+- `Region`: The region in which your Snowflake account is located
+- `Account identifier`: The account identifier for your Snowflake account
+- `Storage integration name`: The name you would like to give the storage integration that will be created
+- `Stage name`: The name you would like to give the stage that will be created
+- `Warehouse name`: The name of the warehouse that you would like to use to import the data
+- `Database name`: The name of the database that you would like to the new table to be created in
+- `Schema name`: The name of the schema that you would like to the new table to be created in
+- `Table name`: The name of the table that will be created
+
+Then click `Run import` and the data will be imported to Snowflake. The data will be imported to the table that you specified in the `Table name` field. If the table already exists, it will be dropped and recreated.
```

Comment:
Should we just error in this case? Worried that someone drops their pipeline or something by accident trying to import data into a live table.

Reply:
Yeah good point. it has been useful when developing to just replace the table each time, but in prod it's dangerous. Will make an issue for this.